internal/pkg/deploy/cloudformation/stack/lb_web_app.go

```diff
@@ -0,0 +1,202 @@
+	if err != nil {
+		return nil, fmt.Errorf("new addons: %w", err)
+	}
+	return &LoadBalancedWebApp{
+		app: &app{
+			name:    mft.Name,
+			env:     env,
+			project: proj,
+			tc:      mft.ApplyEnv(env).TaskConfig,
+			rc:      rc,
+			parser:  parser,
+		},
+		manifest:     mft,
+		httpsEnabled: true,
+
+		parser: parser,
+		addons: addons,
+	}, nil
+}
+
+// Template returns the CloudFormation template for the application parametrized for the environment.
+func (c *LoadBalancedWebApp) Template() (string, error) {
+	rulePriorityLambda, err := c.parser.Read(lbWebAppRulePriorityGeneratorPath)
+	if err != nil {
+		return "", err
+	}
+	outputs, err := c.addonsOutputs()
+	if err != nil {
+		return "", err
+	}
+	content, err := c.parser.ParseAppTemplate(lbWebAppTemplateName, struct {
+		RulePriorityLambda string
+		AddonsOutputs      []addons.Output
+		*lbWebAppTemplateParams
+	}{
+		RulePriorityLambda:     rulePriorityLambda.String(),
+		AddonsOutputs:          outputs,
+		lbWebAppTemplateParams: c.toTemplateParams(),
+	}, template.WithFuncs(map[string]interface{}{
+		"toSnakeCase":           toSnakeCase,
+		"filterSecrets":         filterSecrets,
+		"filterManagedPolicies": filterManagedPolicies,
+	}))
+	if err != nil {
+		return "", err
+	}
+	return content.String(), nil
+}
+
+// Parameters returns the list of CloudFormation parameters used by the template.
+func (c *LoadBalancedWebApp) Parameters() []*cloudformation.Parameter {
+	templateParams := c.toTemplateParams()
+	return append(c.app.Parameters(), []*cloudformation.Parameter{
+		{
+			ParameterKey:   aws.String(LBWebAppContainerPortParamKey),
+			ParameterValue: aws.String(strconv.FormatUint(uint64(templateParams.Image.Port), 10)),
+		},
+		{
+			ParameterKey:   aws.String(LBWebAppRulePathParamKey),
+			ParameterValue: aws.String(templateParams.App.Path),
+		},
+		{
+			ParameterKey:   aws.String(LBWebAppHealthCheckPathParamKey),
+			ParameterValue: aws.String(templateParams.App.HealthCheckPath),
+		},
+		{
+			ParameterKey:   aws.String(LBWebAppHTTPSParamKey),
+			ParameterValue: aws.String(strconv.FormatBool(c.httpsEnabled)),
+		},
+	}...)
+}
+
+// SerializedParameters returns the CloudFormation stack's parameters serialized
+// to a YAML document annotated with comments for readability to users.
+func (c *LoadBalancedWebApp) SerializedParameters() (string, error) {
+	return c.app.templateConfiguration(c)
+}
+
+func (c *LoadBalancedWebApp) addonsOutputs() ([]addons.Output, error) {
+	stack, err := c.addons.Template()
+	if err == nil {
+		return addons.Outputs(stack)
+	}
+
+	var noAddonsErr *addons.ErrDirNotExist
+	if !errors.As(err, &noAddonsErr) {
+		return nil, fmt.Errorf("generate addons template for application %s: %w", c.manifest.Name, err)
+	}
+	return nil, nil // Addons directory does not exist, so there are no outputs and error.
+}
+
+// lbWebAppTemplateParams holds the data to render the CloudFormation template for an application.
+type lbWebAppTemplateParams struct {
```

Comment:
Ideally, I want to remove this struct completely and change our cfn templates to not need it. But these refactors are already getting really big so I wanted to send this one out first.